xwiki-platform-core/xwiki-platform-web/src/main/webapp/templates/rightpanels.vm

```diff
@@ -4,7 +4,8 @@
 ## Global Variable
 #set($xwikiPanelWidth = $rightPanelsWidth)
 #foreach($panelUix in $panelUixs)
-  #if($xwiki.hasAccessLevel('view', $panelUix.getId()))
+  #set($panelDocName = $services.uix.getDocumentName($panelUix))
```

Comment:
>  if not then you should not check right because the name is most probably not a document 

so you mean that checking 
> $xwiki.hasAccessLevel('view', $panelUix.getId())

was wrong in the first place?